.circleci/config.yml

```diff
@@ -12,43 +12,85 @@
 #
 ################################################################################
 
-# Java Maven CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-java/ for more details
-#
 version: 2
 
-shared: &shared
+shared_steps:
+  - &build_all_package
```

Comment:
I think using the "built-in" yaml references syntax/feature is fully correct, but maybe a more "circleCI friendly" way would be to use the "commands" feature introduced by circleCI 2.1 (I used it in https://github.com/adobe/commerce-cif-connector/blob/master/.circleci/config.yml). I don't think it's any better, also because it internally does some inline substitutions, but it might be easier to read and maintain.

Reply:
@cjelger I updated the PR to use the new "command" pattern.